scheduler/src/cook/mesos/api.clj

```diff
@@ -2498,6 +2584,44 @@
                     (->> (pool/all-pools db)
                          (mapv pool-entity->consumable-map))))}))
 
+(def DataLocalUpdateTimeResponse
+  {s/Uuid (s/maybe s/Str)})
+
+(def DataLocalCostResponse
+  {s/Str s/Num})
+
+(defn data-local-update-time-handler
+  "Handler for return the last update time of data locality"
+  [conn]
+  (base-cook-handler
+   {:allowed-methods [:get]
+    :handle-ok (fn [_]
+                 (let [uuid->datasets (->> (d/db conn)
+                                          util/get-pending-job-ents
+                                          (filter (fn [j] (not (empty? (:job/datasets j)))))
```

Comment:
Consider a TODO here? It looks like this is a diagnostic function, but if it is in a critical path, we should be caching uuid->datasets in guava, as it is immutable.

Reply:
Done